R/int_pctl.R

```diff
@@ -137,29 +157,36 @@ boostrap_metrics_by_config <- function(config, seed, x, metrics, times, allow_pa
     cbind(config)
 }
 
-# We have to do the analysis separately for each evaluation time.
-int_pctl_dyn_surv <- function(x, allow_par) {
-  `%op%` <- get_int_p_operator(allow_par)
-  times <- unique(x$.metrics[[1]]$.eval_time)
-  res <-
-    foreach::foreach(
-      i = seq_along(times),
-      .errorhandling = "pass",
-      .packages = c("purrr", "rsample", "dplyr")
-    )  %op% {
-      int_pctl_by_eval_time(times[i], x)
-    }
-  dplyr::bind_rows(res)
+fake_term <- function(x) {
+  x$term <- paste(x$term, format(1:nrow(x)))
+  x
 }
 
-int_pctl_by_eval_time <- function(time, x) {
-  times <- dplyr::tibble(.eval_time = time)
-  x$.metrics <- purrr::map(x$.metrics, ~ dplyr::inner_join(.x, times, by = ".eval_time"))
-  rsample::int_pctl(x, .metrics) %>%
-    dplyr::mutate(.eval_time = time) %>%
-    dplyr::relocate(.eval_time, .after = term)
-}
+# tests in extratests
+# nocov start
+int_pctl_surv <- function(x, allow_par, alpha) {
+  `%op%` <- get_int_p_operator(allow_par)
```

Comment:
Do we still need this? Are you removing parallelization?

Reply:
Yes we do. This recomputed thousands of metrics and does add some computational efficiency.

Reply:
I thought so, I meant more "does this get picked up?" because I do see an explicit call to `foreach::foreach()` in the previous version but in the new version?

Reply:
The changes remove the need for parallelization here; we do it at the previous level of computations.